File: app/src/main/java/com/narsha/wave_android/view/fragment/playlist/PlayListFragment.java
package com.narsha.wave_android.view.fragment.playlist;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;
import androidx.navigation.NavController;
import androidx.navigation.Navigation;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import com.narsha.wave_android.R;
import com.narsha.wave_android.data.RequestUser;
import com.narsha.wave_android.data.response.music.PlayList;
import com.narsha.wave_android.network.Server;
import com.narsha.wave_android.view.adapter.listener.OnPlaylistItemClickListener;
import com.narsha.wave_android.view.adapter.recyclerview.PlaylistAdapter;
import com.narsha.wave_android.viewmodel.MainViewModel;

import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

/**
 * A simple {@link Fragment} subclass.
 * Use the {@link PlayListFragment#newInstance} factory method to
 * create an instance of this fragment.
 */
public class PlayListFragment extends Fragment {

    private MainViewModel model;
    private PlaylistAdapter adapter;
    private Call<List<PlayList>> getMyPlayList;
    private RecyclerView recyclerView;
    private NavController navController;

    public static PlayListFragment newInstance(String param1, String param2) {
        PlayListFragment fragment = new PlayListFragment();
        return fragment;
    }

    private OnPlaylistItemClickListener listener =  (position, playList)->{
        model.playList.setValue(playList);
        navController.navigate(R.id.action_navigation_playlist_to_songListFragment);
    };

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_play_list, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        model = new ViewModelProvider(requireActivity()).get(MainViewModel.class);

        navController = Navigation.findNavController(view);

        recyclerView = view.findViewById(R.id.selectedPlayList);

        // 서버에서 내 플레이리스트 받기
        // 응답 받으면  Adapter 초기화하기 (List<Playlist>, listener)
        // 각 항목이 클릭 되면 SongList로 가기
        RequestUser user= new RequestUser();
        user.setUserid(model.user.getValue().getUserId());
        getMyPlayList = Server.getInstance().getApi().getMyPlayList(user);

        recyclerView.setLayoutManager(new LinearLayoutManager(requireContext()));

        getMyPlayList.enqueue(new Callback<List<PlayList>>() {
            @Override
            public void onResponse(Call<List<PlayList>> call, Response<List<PlayList>> response) {
                if(response.code()==200) {
                    List<PlayList> list = response.body();
                    Log.i("list", "list.size" + list.size() + list.get(1).getTitle());

                    adapter = new PlaylistAdapter(list, listener);
                    recyclerView.setAdapter(adapter);
                } else {
                    Toast.makeText(requireContext(), "오류가 발생했습니다.", Toast.LENGTH_SHORT).show();
                }
            }

            @Override
            public void onFailure(Call<List<PlayList>> call, Throwable t) {
            }
        });
    }
}

File: Wave_V2/app/src/main/java/com/project/wave_v2/view/fragment/PlayListFragment.kt
package com.project.wave_v2.view.fragment

import android.content.Context
import android.content.SharedPreferences
import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import androidx.navigation.NavController
import androidx.navigation.Navigation
import com.project.wave_v2.R
import com.project.wave_v2.data.request.playlist.CallPlayListBody
import com.project.wave_v2.data.response.playlist.MyPlayListModel
import com.project.wave_v2.network.RetrofitClient
import com.project.wave_v2.network.Service
import com.project.wave_v2.view.viewmodel.CallPlayListViewModel
import com.project.wave_v2.view.viewmodel.SearchedViewModel
import com.project.wave_v2.widget.PlayListAdapter
import com.project.wave_v2.widget.sheet.BottomSheet
import kotlinx.android.synthetic.main.fragment_playlist.*
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import retrofit2.Retrofit

class PlayListFragment : Fragment() {

    lateinit var navController: NavController

    var API: Service? = null
    lateinit var retrofit: Retrofit
    lateinit var viewModel: CallPlayListViewModel
    lateinit var viewModelSearch : SearchedViewModel

    var playList = ArrayList<MyPlayListModel>()
    var playListAdapter: PlayListAdapter ?= null


    override fun onPause() {
        super.onPause()
        Log.d("Logd", "PlayListFragment onPause")
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        viewModel = ViewModelProvider(requireActivity()).get(CallPlayListViewModel::class.java)
        viewModelSearch = ViewModelProvider(requireActivity()).get(SearchedViewModel::class.java)


        val prefs: SharedPreferences = requireActivity().getSharedPreferences("user_info", Context.MODE_PRIVATE)
        var id: String? = prefs.getString("userId", "user")

        navController = Navigation.findNavController(view)

        retrofit = RetrofitClient.getInstance()
        API = RetrofitClient.getService()

        playListAdapter = PlayListAdapter(playList, requireContext(), prefs.getString("userId", "user")!!)
        selectedPlayList.adapter = playListAdapter
        selectedPlayList.setHasFixedSize(true)

        callPlayList(id)

        addList_Btn.setOnClickListener {
            var dialog = AddPlayListFragment()
            fragmentManager?.let { it1 -> dialog.show(it1, "addPlayList") }
        }

        with(viewModel) {
            addFinish.observe(viewLifecycleOwner, Observer {
                callPlayList(id)
            })
        }

    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_playlist, container, false)
    }


    private fun callPlayList(id: String?) {
        API?.myList(CallPlayListBody(userId = id))
                ?.enqueue(object : Callback<List<MyPlayListModel>> {
                    override fun onResponse(call: Call<List<MyPlayListModel>>, response: Response<List<MyPlayListModel>>) {
                        playList.clear()
                        for (i in 0 until response.body()?.size!!) {
                            playList.add(0, response.body()!![i])
                        }
                        playListAdapter!!.notifyDataSetChanged()
                    }

                    override fun onFailure(call: Call<List<MyPlayListModel>>, t: Throwable) {
                    }


                })
    }
}


